book.adoc

```diff
@@ -14,11 +14,34 @@ Antoine Sabot-Durand <antoine@sabot-durand.net>
 = Part 1 First step with CDI
 
 == Introduction to CDI
+=== Dependency injection
+* The organization of classes, dependencies between classes
+* Inheritance vs delegation, collaboration
+* Why dependency injection got some attention
+* Some pitfalls that make DI impossible to use (inh vs deleg)
+* The question of testing : dependency injection makes the use of mocks so easy
+* Dependency injection everywhere 
+* Three ways to inject objects
+* Discussion
+
 === What is CDI?
+* Injection frameworks in Java (a little history : Fortress, Pico container, Hivemind, Spring)
+* Dependency injection useful in the Java EE world, Java SE left aside (for now)
+* Managed environment vs unmanaged : the need for a special object to handle the building of objects
+* A first notion of container
+* Injecting means producing
+
 === Problems solved by CDI
+* Injection, interception, decoration, events : back to the roots
+
 === The CDI specification
-=== The CDI implementations
+* Talk a little about the way specifications appear in the Java EE world
```

Comment:
The ide of this part is also to explain what is the JCP, a specification, Java EE, etc... but from a CDI pov